python/social_bot/tasks.py

```diff
@@ -290,6 +369,12 @@ def _move_goal(self, goal, agent_loc):
         while True:
             loc = (random.random() * range - range / 2,
                    random.random() * range - range / 2, 0)
+            if self._also_curriculum_target_angle and not self._is_full_range_in_curriculum:
+                dist = random.random() * range
+                angle = math.radians(
+                    math.degrees(agent_dir[2]) +
+                    random.random() * self._random_angle - self._random_angle / 2)
+                loc = (dist * math.cos(angle), dist * math.sin(angle), 0) + agent_loc
```

Comment:
@JiaLn is right. Here the maximum distance is range. It's range/sqrt(2) at line 370. And line 370 put higher probability for large distance.